DSharpPlus.Lavalink/LavalinkNodeConnection.cs

```diff
@@ -131,6 +131,26 @@ public sealed class LavalinkNodeConnection
         }
         private readonly AsyncEvent<LavalinkGuildConnection, TrackExceptionEventArgs> _trackException;
 
+        /// <summary>
+        /// Triggered whenever a new guild connection is created.
+        /// </summary>
+        public event AsyncEventHandler<LavalinkGuildConnection, AsyncEventArgs> GuildConnectionCreated
+        {
+            add { this._guildConnectionCreated.Register(value); }
+            remove { this._guildConnectionCreated.Unregister(value); }
+        }
+        private readonly AsyncEvent<LavalinkGuildConnection, AsyncEventArgs> _guildConnectionCreated;
+
+        /// <summary>
+        /// Triggered whenever a guild connection is removed.
+        /// </summary>
+        public event AsyncEventHandler<LavalinkGuildConnection, AsyncEventArgs> GuildConnectionRemoved
+        {
+            add { this._guildConnectionRemoved.Register(value); }
+            remove { this._guildConnectionRemoved.Unregister(value); }
+        }
+        private readonly AsyncEvent<LavalinkGuildConnection, AsyncEventArgs> _guildConnectionRemoved;
```

Comment:
Not sure I like throwing `AsyncEventArgs` here. Is it not possible to give a subtype here?

Reply:
That's totally fair. I wasn't sure what else to put here, since this actually doesn't need or take any arguments. Would you want me to make an extended class of it extending AsyncEventArgs with no extra changes just to have that type, or is there an alternate solution that would be better?

Reply:
Probably.  Sounds good.